volatility3/framework/plugins/windows/psxview.py

```diff
@@ -0,0 +1,279 @@
+                                    handles_plugin.get_type_map(
+                                        self.context, layer_name, symbol_table
+                                    )
+                                )
+                                == "Process"
+                            ):
+                                ret.append(h.Body.cast("_EPROCESS"))
+
+                except exceptions.InvalidAddressException:
+                    vollog.log(
+                        constants.LOGLEVEL_VVV, "Cannot access eprocess object table"
+                    )
+
+        ret = self.filter_garbage_procs(ret)
+        return self.proc_list_to_dict(ret)
+
+    def check_session(self, pslist_procs):
+        procs = [p for p in pslist_procs if p.get_session_id() != None]
+
+        return self.proc_list_to_dict(procs)
+
+    def _generator(self):
+        kernel = self.context.modules[self.config["kernel"]]
+
+        layer_name = kernel.layer_name
+        symbol_table = kernel.symbol_table_name
+
+        kdbg_list_processes = list(
+            pslist.PsList.list_processes(
+                context=self.context, layer_name=layer_name, symbol_table=symbol_table
+            )
+        )
+
+        processes = {}
+
+        processes["pslist"] = self.check_pslist(kdbg_list_processes)
+        processes["psscan"] = self.check_psscan(layer_name, symbol_table)
+        processes["thrdscan"] = self.check_thrdscan()
+        processes["csrss"] = self.check_csrss_handles(
+            kdbg_list_processes, layer_name, symbol_table
+        )
+        processes["sessions"] = self.check_session(kdbg_list_processes)
+
+        seen_offsets = set()
+        for source in processes:
+            for offset in processes[source]:
+                if offset not in seen_offsets:
+                    seen_offsets.add(offset)
+                    proc = processes[source][offset]
+
+                    pid = proc.UniqueProcessId
+                    name = self.proc_name_to_string(proc)
+
+                    exit_time = proc.get_exit_time()
+                    if type(exit_time) != datetime.datetime:
+                        exit_time = ""
+                    else:
+                        exit_time = str(exit_time)
+
+                    in_sources = {
+                        src: str(offset in processes[src]) for src in processes
+                    }
+
+                    if self.config["identify-expected"]:
+                        f = "False"
+                        n = "Normal"
+
+                        if in_sources["pslist"] == f:
+                            if exit_time != "":
+                                in_sources["pslist"] = n
+
+                        if in_sources["thrdscan"] == f:
+                            if exit_time != "":
+                                in_sources["thrdscan"] = n
+
+                        if in_sources["csrss"] == f:
+                            if name.lower() in ["system", "smss.exe", "csrss.exe"]:
```

Comment:
This hardcoded list of processes that are guaranteed safe is a little worrying...  Why are processes with these names safe, what's to stop malware naming itself this way and getting marked as normal?

Reply:
This is something I didn't feel great about either. I copied it from the original Psxview, but to be honest I don't think the feature should be in the plugin at all, for the same reasons you mentioned - it could easily be misleading.